### PR TITLE
ddl,sinks: rename Kafka setting exactly_once to reuse_topic

### PR DESCRIPTION
**Pull request by aljoscha**
This better describes the user-facing effect and will make it easier to
document.

We now refer to the setting as reuse_topic up until
KafkaSinkConnectorBuilder. Internal code, starting with
KafkaSinkConnector refers to it as exactly_once, which is still the
underlying implementation mechanism.

### CODE REVIEW

**Comment by aljoscha** [comment]
This unblocks #7390

File: src/sql/src/plan/statement/ddl.rs [hunk 1]
```diff
@@ -1278,30 +1278,30 @@ fn kafka_sink_builder(
...
     }
 
-    let transitive_source_dependencies: Vec<_> = if exactly_once {
+    let transitive_source_dependencies: Vec<_> = if reuse_topic {
         for item in root_dependencies.iter() {
             if item.item_type() == CatalogItemType::Source {
                 if !item.source_connector()?.yields_stable_input() {
                     bail!(
-                    "all input sources of an exactly-once Kafka sink must be replayable, {} is not",
+                    "reuse_topic makes a sink an exactly-once sink, all input sources of an exactly-once Kafka sink must be replayable, {} is not",
```
Comment (ruchirK) [inline src/sql/src/plan/statement/ddl.rs:1296]: I think we should reword this because this error message begs the question "well wtf is an exactly-once sink?"

Maybe we can reword as:

"reuse_topic requires that sink input dependencies are replayable, {} is not"

File: src/dataflow-types/src/types.rs [hunk 2]
```diff
@@ -1138,7 +1138,10 @@ pub struct KafkaSinkConnectorBuilder {
     pub consistency_value_schema: Option<String>,
     pub config_options: BTreeMap<String, String>,
     pub ccsr_config: ccsr::ClientConfig,
-    pub exactly_once: bool,
+    // This makes the sink an exactly_one sink because we need to ensure
```
Comment (ruchirK) [inline src/dataflow-types/src/types.rs:1141]: I think we should just get rid of the exactly_once terminology both in internal and user facing documentation. So we could reword this to

```
Forces the sink to always write to the same topic across restarts instead of picking a new topic each time.
```

**Review by ruchirK — APPROVED** [review]
LGTM! left some documentation nits

File: src/sql/src/plan/statement/ddl.rs (hunk 1 above)
Comment (aljoscha) [reply src/sql/src/plan/statement/ddl.rs:1296]: 👍

File: src/dataflow-types/src/types.rs (hunk 2 above)
Comment (aljoscha) [reply src/dataflow-types/src/types.rs]: 👍

**Comment by aljoscha** [comment]
Thanks for the review! I changed according to your suggestions and will merge once we're green.

**Review by msudberg — APPROVED** [review]
I can't comment on the code, but the naming lgtm. :)
